### PR TITLE
Clean up needsParens

### PR DESCRIPTION
**Pull request by bakkot**
After #921, I noticed needsParens has a lot of weird fallthrough. This eliminates or makes explicit all of it.

This cleanup changes formatting in some cases: what was previously
```js
+x + y;
(+x) * y;
```
is now

```js
+x + y;
+x * y;
```

### CODE REVIEW

File: tests/unary_expression/urnary_expression.js [hunk 1]
```diff
@@ -4,3 +4,5 @@ x--;
 -+1;
 x + + + + 1;
 x + (+ (+ (+ 1)))
+x * +y;
++x * y;
```
Comment (vjeux) [inline tests/unary_expression/urnary_expression.js]: can you add an empty line :)

**Comment by vjeux** [comment]
Thanks! could you just add a new line and i'll merge it

File: tests/unary_expression/urnary_expression.js (hunk 1 above)
Comment (bakkot) [inline tests/unary_expression/urnary_expression.js]: Before or after the tests?

Comment (vjeux) [inline tests/unary_expression/urnary_expression.js]: I mean, a \n at the end of the file, to remove the little red `(/)<-` icon on github ui

Comment (yamafaktory) [inline tests/unary_expression/urnary_expression.js]: @bakkot in the end of `tests/unary_expression/urnary_expression.js` (by the way, there's a typo).

Comment (yamafaktory) [inline tests/unary_expression/urnary_expression.js]: Oh sorry, duplicate...

Comment (bakkot) [inline tests/unary_expression/urnary_expression.js]: Ah, done.

@yamafaktory, lol. Not gonna fix it here I guess.

File: tests/unary_expression/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -7,12 +7,15 @@ x--;
 -+1;
 x + + + + 1;
 x + (+ (+ (+ 1)))
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+x * +y;
++x * y;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (vjeux) [inline tests/unary_expression/__snapshots__/jsfmt.spec.js.snap]: you also need to regenerate the tests

Comment (bakkot) [inline tests/unary_expression/__snapshots__/jsfmt.spec.js.snap]: Whoops... done.

**Pull request merged**